core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java

```diff
@@ -109,15 +110,17 @@ private Scanner executeQueryWithIndex(Selection selection, TableMetadata tableMe
     ValueBinder binder = new ValueBinder(DynamoOperation.VALUE_ALIAS);
     keyColumn.accept(binder);
     Map<String, AttributeValue> bindMap = binder.build();
-    builder
-        .keyConditionExpression(condition)
-        .expressionAttributeValues(bindMap)
-        .expressionAttributeNames(ImmutableMap.of(expressionColumnName, column));
+    builder.keyConditionExpression(condition).expressionAttributeValues(bindMap);
+
+    Map<String, String> expressionAttributeNames = new HashMap<>();
+    expressionAttributeNames.put(expressionColumnName, column);
 
     if (!selection.getProjections().isEmpty()) {
-      projectionExpression(builder, selection);
+      projectionExpression(builder, selection, expressionAttributeNames);
     }
 
+    builder.expressionAttributeNames(expressionAttributeNames);
```

Comment:
In this fix, we first merge the `expressionAttributeNames` and set it to the builder.